### DISCUSSION TITLE
LDAPConnection.Search doesn't get repeated attribute names

### DISCUSSION THREAD

**Discussion by JeremyGEA** [dsbenghe/Novell.Directory.Ldap.NETStandard#223] [General]
Our LDAP server has many 'memberOf' membership attributes and other attributes that repeat the same attribute name and this can be seen using LDAPAdmin or other means. Where I should be seeing ~135 attributes, this library is only pulling 91. It seems to be only grabbing the first attribute of each unique name. For example, where I have 31 'memberOf' roles, it only shows 1 'memberOf' attributes in the entity's AttributeSet.

I'm using version 3.6.0.

**Comment by Mcheimech** [comment]
You need to make sure you're retrieving all attributes by passing the array parameter attrs as below:

`ldapConnection.Search(ContainerDN, LdapConnection.ScopeSub, EntryCN, new[] { LdapConnection.AllUserAttrs }, false);`

**Comment by dsbenghe** [comment]
Likely similar with this https://github.com/dsbenghe/Novell.Directory.Ldap.NETStandard/issues/1

Use StringValues?
